crates/turborepo-lib/src/run/mod.rs

```diff
@@ -364,8 +366,30 @@ impl<'a> Run<'a> {
 
         let scm = SCM::new(&self.base.repo_root);
 
-        let filtered_pkgs =
-            scope::resolve_packages(&opts.scope_opts, &self.base.repo_root, &pkg_dep_graph, &scm)?;
+        let filtered_pkgs = {
+            let mut filtered_pkgs = scope::resolve_packages(
+                &opts.scope_opts,
+                &self.base.repo_root,
+                &pkg_dep_graph,
+                &scm,
+            )?;
+
+            if filtered_pkgs.len() != pkg_dep_graph.len() {
+                for target in self.targets() {
+                    let task_name = TaskName {
+                        package: Some(ROOT_PKG_NAME.into()),
+                        task: target.into(),
+                    };
+
+                    if root_turbo_json.pipeline.contains_key(&task_name) {
```

Comment:
`TurboJson` has a private method `has_task` that I think does what we want, I think making it public and using it here might make sense instead of trying to construct the right `TaskName` for lookup

Reply:
I'm not sure that's what we want. Say we have a target `build`, `has_task` will be true if we have a task definition for `build` in the pipeline. What we really want is for it to be true only if there is a task definition for `//#build` in the pipeline, since that will override the default behavior and include the root workspace in the filtered pkgs.